ext/io/12_io.js

```diff
@@ -64,6 +68,11 @@ async function* iter(
   r,
   options,
 ) {
+  internals.warnOnDeprecatedApi(
+    "Deno.iter()",
+    new Error().stack,
+    "Use `ReadableStreamDefaultReader` instead.",
```

Comment:
Err I don't understand this suggestion :( don't we have `iter()` equivalent in `deno_std`?

Reply:
No, that's deprecated. Should we add it back in?

Reply:
Probably not. Maybe we shouldn't add a suggestion here - this API relies on `Reader` interface that is also deprecated. So migrating the underlying `Reader` dictates which API should be used instead.

Reply:
The `Reader` interface in `deno_std` is no longer deprecated, so this might be fine to add back in.

Also, actually, should it not be `ReadableStream` instead? Both `ReadableStream` and `Deno.iter()` are async iterable.

We can either add `iter()` back to `deno_std` or suggest `ReadableStream`.

Reply:
Let's suggest `ReadableStream` 👍